pennylane_orquestra/cli_actions.py

```diff
@@ -0,0 +1,231 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains utilities and auxiliary functions for using the Orquestra
+Quantum Engine command line interface (CLI).
+"""
+import subprocess
+import time
+import os
+import urllib.request
+import json
+import tarfile
+
+import yaml
+from appdirs import user_data_dir
+
+
+def qe_get(workflow_id, option="workflow"):
+    """Function for getting information via an Orquestra Quantum Engine CLI
+    call.
+
+    This function is mostly used for retrieving workflow related information.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Kwargs:
+        option (str): The option specified for the ``qe get`` CLI call.
+            Examples include ``workflow`` and ``workflowresult``.
+
+    Returns:
+        str: response message of the CLI call
+    """
+    process = subprocess.Popen(
+        ["qe", "get", option, str(workflow_id)], stdout=subprocess.PIPE, universal_newlines=True
+    )
+    return process.stdout.readlines()
+
+
+def qe_submit(filepath, keep_file=False):
+    """Function for submitting a workflow via a CLI call.
+
+    Handling submission response messages was based on using Orquestra API
+    v1.0.0.
+
+    Args:
+        filepath (str): the filepath for the workflow file
+
+    Keyword Args:
+        keep_file=False (bool): whether or not to keep or delete the workflow
+            file after submission
+
+    Returns:
+        str: the ID of the workflow submitted
+
+    Raises:
+        ValueError: if the submission was not successful
+    """
+    process = subprocess.Popen(
+        ["qe", "submit", "workflow", str(filepath)], stdout=subprocess.PIPE, universal_newlines=True
+    )
+    res = process.stdout.readlines()
+
+    details = "".join(res)
+    if "Success" not in details:
+        raise ValueError(res)
+
+    if not keep_file:
+        os.remove(filepath)
+
+    unexpected_resp_msg = "Received an unexpected response after submitting workflow."
+    if isinstance(res, list):
+        try:
+            # Get the workflow ID after submitting a workflow
+            workflow_id = res[1].split()[-1]
+        except IndexError as e:
+            raise ValueError(unexpected_resp_msg) from e
+    else:
+        raise ValueError(unexpected_resp_msg)
+
+    return workflow_id
+
+
+def workflow_details(workflow_id):
+    """Function for getting workflow details via a CLI call.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Returns:
+        str: response message of the CLI call
```

Comment:
Depending on the answer to my questoin above, you might also need to update the return type here